packages/components/src/color-palette/index.native.js

```diff
@@ -0,0 +1,215 @@
+/**
+ * External dependencies
+ */
+import {
+	ScrollView,
+	TouchableWithoutFeedback,
+	View,
+	Animated,
+	Easing,
+} from 'react-native';
+import { map } from 'lodash';
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect, createRef } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { withPreferredColorScheme } from '@wordpress/compose';
+/**
+ * Internal dependencies
+ */
+import styles from './style.scss';
+import ColorIndicator from '../color-indicator';
+
+const ANIMATION_DURATION = 200;
+
+function ColorPalette( {
+	setBackgroundColor,
```

Comment:
Hmmm, could we simplify it a bit? WDYT about changing `setBackgroundColor`, `setTextColor` and `setGradient` to one prop called `onColorSelect`. We could also remove the `currentScreen` and change the `backgroundColor` and `textColor` to the `color`. etc...